// google-books-js/dist/api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleBooksAPI = void 0;
class GoogleBooksAPI {
    BASE_URL = "https://www.googleapis.com/books/v1/volumes";
    async search(params) {
        const response = await fetch(`${this.BASE_URL}?q=${this.formatSearchFiltersToQueryString(params.filters)}`);
        return (await response.json());
    }
    formatSearchFiltersToQueryString(filters) {
        const mappedFields = {
            title: "intitle",
            author: "inauthor",
            publisher: "inpublisher",
            subject: "subject",
            isbn: "isbn",
        };
        return Object.keys(filters)
            .filter((param) => filters[param])
            .map((param) => `${mappedFields[param]}:${filters[param]}`)
            .join("&");
    }
    async getVolume(volumeId) {
        const response = await fetch(`${this.BASE_URL}/${volumeId}`);
        const data = (await response.json());
        if (data.error) {
            throw Error("Volume not found");
        }
        return data;
    }
}
exports.GoogleBooksAPI = GoogleBooksAPI;
